src/objectcontroller.cpp

```diff
@@ -1478,6 +1484,299 @@ void DateObject::getDate(int *day, int *month, int *year)
+    else
+    {
+        val->getDateTimeValue(&day, &month, &year, &wday, &hour, &min, &sec);
+        if ( day_m != day || month_m != month || year_m != year ||
+	     wday_m != wday || hour_m != hour || min_m != min || sec_m != sec )
+        {
+            day_m = day;
+            month_m = month;
+            year_m = year;
+            wday_m = wday;
+            hour_m = hour;
+            min_m = min;
+            sec_m = sec;
+            return true;
+        }
+    }
+    return false;
+}
+
+bool DateTimeObjectValue::set(double value)
+{
+    int date, day, month, year, wday, hour, min, sec;
+    if (value < 0)
+    {
+        day = -1;
+        month = -1;
+        year = -1;
+        wday = -1;
+        hour = -1;
+        min = -1;
+        sec = -1;
+    }
+    else
+    {
+	date = value / 86400;
+	value -= date * 86400;
+
+        year = date / 400;
+        date -= date * 400;
+        month = date / 31;
+        day = date - month * 31;
+
+	wday = 0;
+
+	hour = value / 3600;
+	value -= hour * 3600;
+	min = value / 60;
+	sec = value - min * 60;
+    }
+    if ( day_m != day || month_m != month || year_m != year|| wday_m != wday ||  hour_m != hour || min_m != min || sec_m != sec )
+    {
+        day_m = day;
+        month_m = month;
+        year_m = year;
+        wday_m = wday;
+        hour_m = hour;
+        min_m = min;
+        sec_m = sec;
+        return true;
+    }
+    return false;
+}
+
+void DateTimeObjectValue::getDateTimeValue(int *day, int *month, int *year, int* wday, int *hour, int *min, int *sec)
+{
+    if (day_m == -1)
+    {
+        time_t t = time(0);
+        struct tm * timeinfo = localtime(&t);
+        *day = timeinfo->tm_mday;
+        *month = timeinfo->tm_mon+1;
+        *year = timeinfo->tm_year;
+        *wday = timeinfo->tm_wday;
+	if (*wday == 0)
+          *wday = 7;
+        *hour = timeinfo->tm_hour;
+        *min = timeinfo->tm_min;
+        *sec = timeinfo->tm_sec;
+    }
+    else
+    {
+        *day = day_m;
+        *month = month_m;
+        *year = year_m;
+        *wday = wday_m;
+        *hour = hour_m;
+        *min = min_m;
+        *sec = sec_m;
+    }
+}
+
+Logger& DateTimeObject::logger_m(Logger::getInstance("DateTimeObject"));
+
+DateTimeObject::DateTimeObject() : DateTimeObjectValue(0, 0, 0, 0, 0, 0, 0)
+{}
+
+DateTimeObject::~DateTimeObject()
+{}
+
+ObjectValue* DateTimeObject::createObjectValue(const std::string& value)
+{
+    return new DateTimeObjectValue(value);
+}
+
+void DateTimeObject::setValue(const std::string& value)
+{
+    DateTimeObjectValue val(value);
+    Object::setValue(&val);
+}
+
+void DateTimeObject::doWrite(const uint8_t* buf, int len, eibaddr_t src)
+{
+    if (len < 10)
+    {
+        logger_m.errorStream() << "Invalid packet received for DateTimeObject (too short)" << endlog;
+        return;
+    }
+    int day, month, year, wday, hour, min, sec;
+
+    year = buf[2];
+    month = buf[3];
+    day = buf[4];
+    wday = (buf[5] & 0xE0) >> 5;
```

Comment:
@korni 
You said in the description:

> wday_m is not set/checked when received from the bus, only when set by localtime

but isn't that line actually taking the weekday into account? Otherwise, can you explain?

Reply:
Yes, it is received from the bus and put into m_wday, but there are no consistency checks. In principle, wday should be calendar-consistent with the date. I think this is a shortcoming of the KNX spec that allows for contradictions between the date part and the workday. For linknx it depends on whether we are the authoritative source for datetimes, or are just receiving one, if we should correct inconsistencies. 
This is somewhat similar in TimeObject: wday is set to 0 in most internal cases, to the actual wday for "now" and copied as-is from the bus.
BTW, ETS interprets wday==0 as "everyday" - the DP is also used for time ranges, so there are a few use cases that are not covered. Also, the time "24:00:00" is valid and used as an range endpoint.

Reply:
> Have you had a look at test/ObjectTest.cpp
Now I have,. A test is provided and passes for me. I had to change the String format to make it parsable. The 'T' was used according to ISO 8601.